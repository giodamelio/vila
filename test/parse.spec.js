const path = require('path');

const OpenAPI = require('../src/parse');

describe('Parse OpenAPI specs', () => {
  describe('fromFile()', () => {
    it('Parse YAML spec', async () => {
      expect.assertions(1);

      const spec = await OpenAPI.fromFile(
        path.join(__dirname, 'openapi_specs/simple.yaml')
      );

      expect(spec).toBeDefined();
    });

    it('Parse JSON spec', async () => {
      expect.assertions(1);

      const spec = await OpenAPI.fromFile(
        path.join(__dirname, 'openapi_specs/simple.json')
      );

      expect(spec).toBeDefined();
    });

    it('Spec file does not exist', async () => {
      expect.assertions(1);

      await expect(
        OpenAPI.fromFile('this file does not exist')
      ).rejects.toThrowErrorMatchingInlineSnapshot(
        `"ENOENT: no such file or directory, open 'this file does not exist'"`
      );
    });

    it('Version must match OpenAPI 3.x', async () => {
      expect.assertions(2);

      await expect(
        OpenAPI.fromFile(
          path.join(__dirname, 'openapi_specs/swagger_version_2.json')
        )
      ).rejects.toThrowErrorMatchingInlineSnapshot(
        `"Input must be an OpenAPI spec"`
      );
      await expect(
        OpenAPI.fromFile(
          path.join(__dirname, 'openapi_specs/openapi_version_4.json')
        )
      ).rejects.toThrowErrorMatchingInlineSnapshot(
        `"Input must be an OpenAPI whose version matches 3.x"`
      );
    });
  });
});
