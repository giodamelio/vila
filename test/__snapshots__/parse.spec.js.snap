// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse OpenAPI specs constructor() Propertly insert routes into database 1`] = `
Array [
  Object {
    "operationId": "echo",
    "requestBody": Object {
      "content": Object {
        "application/json": Object {
          "example": "{
  \\"message\\": \\"Hello Echo Server!\\"
}
",
          "schema": Object {
            "$ref": "#/components/schemas/EchoBody",
          },
        },
      },
      "description": "Message with some additional info from the server",
      "required": true,
    },
    "responses": Object {
      "200": Object {
        "content": Object {
          "application/json": Object {
            "example": "{
  \\"message\\": \\"Hello Echo Server!\\",
  \\"length\\": 18
}
",
            "schema": Object {
              "$ref": "#/components/schemas/EchoResponse",
            },
          },
        },
        "description": "Successfull echo",
      },
      "400": Object {
        "content": Object {
          "application/json": Object {
            "example": "{
  \\"error\\": \\"No message sent!\\"
}
",
            "schema": Object {
              "$ref": "#/components/schemas/EchoError",
            },
          },
        },
        "description": "No message sent",
      },
    },
    "summary": "Echo a string back",
  },
]
`;

exports[`Parse OpenAPI specs getRouteByOperationId() Get a route by its operation id 1`] = `
Object {
  "operationId": "echo",
  "requestBody": Object {
    "content": Object {
      "application/json": Object {
        "example": "{
  \\"message\\": \\"Hello Echo Server!\\"
}
",
        "schema": Object {
          "$ref": "#/components/schemas/EchoBody",
        },
      },
    },
    "description": "Message with some additional info from the server",
    "required": true,
  },
  "responses": Object {
    "200": Object {
      "content": Object {
        "application/json": Object {
          "example": "{
  \\"message\\": \\"Hello Echo Server!\\",
  \\"length\\": 18
}
",
          "schema": Object {
            "$ref": "#/components/schemas/EchoResponse",
          },
        },
      },
      "description": "Successfull echo",
    },
    "400": Object {
      "content": Object {
        "application/json": Object {
          "example": "{
  \\"error\\": \\"No message sent!\\"
}
",
          "schema": Object {
            "$ref": "#/components/schemas/EchoError",
          },
        },
      },
      "description": "No message sent",
    },
  },
  "summary": "Echo a string back",
}
`;
